aten/src/ATen/native/cpu/Reduce.h

```diff
@@ -158,6 +160,27 @@ struct all_same : guts::conjunction<
   std::is_same<T, Args>...
 > {};
 
+template <typename T, typename acc_t, bool is_welford>
+struct welford_inner_reduce {
+  static inline void call(T * data, int64_t size, acc_t & acc) {}
+};
+
+template <typename T, typename acc_t>
+struct welford_inner_reduce<T, acc_t, true> {
+  static inline void call(T * data, int64_t size, acc_t & acc) {
+    double new_mean = 0, new_m2 = 0;
+    std::tie(new_mean, new_m2) = RowwiseMoments(data, size);
+    new_m2 = new_m2 * size;
+    double delta = new_mean - acc.mean;
+    double new_count = acc.nf + size;
+    double nb_over_n = double(size) / new_count;
+    acc.mean = acc.mean + delta * nb_over_n;
+    acc.m2 = acc.m2 + new_m2 + delta * delta * acc.nf * nb_over_n;
```

Comment:
This doesn't seem to be correct. Instead of `delta * delta`, we need `delta * (new_mean - acc.mean)` where the `acc.mean` is the updated value here.

Reply:
Update acc.m2 and acc.mean following the `combine` method of `WelfordOps` https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/SharedReduceOps.h#L114. This PR is just a workaround, and we will improve the performance and precision for reduction kernel utilizing tensoriterator.

Reply:
OK, but I don't think the code you referred to is correct either. Check the algorithm here: https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance

It is something we should fix.

Reply:
The `combine` method of `WelfordOps` also refers to the parallel Welford's online algorithm of https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance .

Reply:
Can you double check its correctness as I pointed out?

Reply:
Yes, and I checked the results using `combine` method of WelfordOps vs numpy as what https://github.com/pytorch/pytorch/blob/master/test/test_reductions.py#L2575 does.